src/Microsoft.DotNet.Wpf/src/Shared/MS/Win32/SafeNativeMethodsOther.cs

```diff
@@ -124,23 +124,24 @@ public static int GetCaretBlinkTime()
         public const UInt16 C3_IDEOGRAPH  = 0x0100;
         public const UInt16 C3_KASHIDA    = 0x0200;
 
-        public static bool GetStringTypeEx(uint locale, uint infoType, char[] sourceString, int count,
-            UInt16[] charTypes)
+        public static unsafe bool GetStringTypeEx(uint locale, uint infoType, ReadOnlySpan<char> sourceString, ReadOnlySpan<UInt16> charTypes)
         {
-            bool win32Return = SafeNativeMethodsPrivate.GetStringTypeEx(locale, infoType, sourceString, count, charTypes);
-            int win32Error = Marshal.GetLastWin32Error();
-
-            if (!win32Return)
+            // Since we do not use [LibraryImport], Span<T> marshallers are not available by default
+            fixed (char* ptrSourceString = sourceString)
             {
-                throw new Win32Exception(win32Error);
-            }
-
-            return win32Return;
+                fixed (ushort* ptrCharTypes = charTypes)
```

Comment:
Nit: You could put `ptrCharTypes` next to `ptrSourceString` to reduce indentation. Like this:
```
fixed (char* ptrSourceString = sourceString)
fixed (ushort* ptrCharTypes = charTypes)
{
    ...
}
```